gcp/modules/locust/swarm.tf

```diff
@@ -0,0 +1,100 @@
+resource "null_resource" "locust_swarm_session" {
+  depends_on = ["module.locust"]
+
+  triggers = {
+    nonce = "${var.nonce}"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOF
+      RETRIES=10
+      RETRY_COUNT=1
+      WORKERS_READY=0
+      while [ "$WORKERS_READY" -lt "${var.locust_workers}" ]; do
+        echo "[Try $RETRY_COUNT of $RETRIES] Waiting for all Locust workers to join the master..."
+        WORKERS_READY=$(kubectl -n locust logs deployment/locust-master --tail 1 | grep -oE "Currently \d+ clients" | grep -oE "\d+")
+        if [ "$WORKERS_READY" == "" ]; then
+          WORKERS_READY=0
+        fi
+        echo "Number of ready workers: $WORKERS_READY out of ${var.locust_workers}!"
+        RETRY_COUNT=$(($RETRY_COUNT+1))
+        if [ "$RETRY_COUNT" -eq "$RETRIES" ] ; then
+          echo "Retry limit reached, giving up!"
+          exit 1
+        fi
+        sleep 10
+      done
+
+      LOCUST_URL=http://127.0.0.1:8089
+      kubectl -n locust port-forward deployment/locust-master 8089:8089 </dev/null &>/dev/null &
```

Comment:
How about putting this command in a variable, so we can `pkill` it precisely later?

Reply:
Okay.